debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresConnectorTask.java

```diff
@@ -151,11 +163,18 @@ else if (snapshotter.shouldStream()) {
     }
 
     @Override
-    public void commit() throws InterruptedException {
+    public void commitRecord(SourceRecord record) throws InterruptedException {
         if (running.get()) {
-            if (lastCompletelyProcessedLsn != null) {
-                producer.commit(lastCompletelyProcessedLsn);
-                logger.info("[DEBEZIUM_DATA_DEBUG] COMMIT {}", LogSequenceNumber.valueOf(lastCompletelyProcessedLsn));
+            Long completedLsn = null;
+            synchronized (this) {
+                if (batchLsns.size() > 0 && batchLsns.get(0).offset.equals(record.sourceOffset())) {
+                    completedLsn = batchLsns.get(0).lsn;
+                    batchLsns.remove(0);
+                }
+            }
+            if (completedLsn != null) {
+                logger.error("[DEBEZIUM_DATA_DEBUG] COMMIT " + LogSequenceNumber.valueOf(completedLsn), new Exception());
+                producer.commit(completedLsn);
```

Comment:
If we wanted to make this "more efficient", I think we could move this conditional block `if (completedLsn != null)` to `commit()`, and leave the other logic to set `completedLsn` in `commitRecord`

Reply:
This is going to change, I'll submit another pull request.

Reply:
(edit: never mind, didn't read your comment correctly) - as far as commit vs commitRecord - according to the description from Confluent, you should only need to implement one or the other. So I was going to stay in sync with that behavior. But if you think it makes more sense to split it up, we can do that too.

Reply:
I'm on the fence. I could see making a bunch of `flushLsn` calls adding some load. I doubt it'd matter much for us, but it might for someone else if we PR it back.

On the other hand, both methods are already being called and I have a hard time seeing Confluent changing that. From a practical point of view, I'm not even sure how they would enforce something like "don't call method B if method A is implemented". It's a good opportunity for us to be a bit more efficient.

We could also just leave this as is, but leave a comment on the PR we open against Debezium.

Reply:
This only forwards the commit once per batch - there is no difference in load with previous behavior in that sense.

Reply:
Got it - I understand now! Ignore my comment. I think this is reasonable.